spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/DataAddress.java

```diff
@@ -70,10 +70,24 @@ public void setKeyName(String keyName) {
         properties.put(KEY_NAME, keyName);
     }
 
+    public Builder toBuilder() {
+        return new Builder(copy());
+    }
+
+    private DataAddress copy() {
+        return Builder.newInstance()
+                .properties(this.properties)
```

Comment:
this is a shallow copy and makes changes to one of the copies affect the other

Reply:
Wouldn't you consider this a deep copy? As the map only contains strings, those all should be immutable. A reference to the original map is not stored.

```java
        public Builder properties(Map<String, String> properties) {
            properties.forEach(this::property);
            return this;
        }

        public Builder property(String key, String value) {
            Objects.requireNonNull(key, "Property key null.");
            address.properties.put(key, value);
            return this;
        }
```

Reply:
Correct, my mistake.

We're missing UTs for the new methods though

Reply:
Thanks for the reminder! I have added unit tests.

Doing this I was able to find and fix an existing bug.